### PR TITLE
fix(babel-types): accept `UnaryExpression` in `TSLiteralType`

### LINKED ISSUE #13327: [Bug]: TSLiteralType should accept UnaryExpression
### 💻

- [ ] Would you like to work on a fix?

### How are you using Babel?

Programmatic API (`babel.transform`, `babel.parse`)

### Input code

See REPL: https://runkit.com/alexandre-cai/60a2ae9feaab4400130133c8

```js
transform(`
  type Test = 1 | -1;
`, {
  parserOpts: {
    plugins: ["typescript"],
  },
  plugins: [
    ({ types: t }) => ({
      visitor: {
        TSLiteralType(path) {
          path.node = t.tsLiteralType(path.node.literal)
        }
      }
    }),
  ],
})
```

### Configuration file name

_No response_

### Configuration

_No response_

### Current and expected behavior

Babel currently accepts UnaryExpression while parsing TSLiteralType, but can’t generate one. Output: 
```
TypeError: unknown: Property literal of TSLiteralType expected node to be of a type 
["NumericLiteral","StringLiteral","BooleanLiteral","BigIntLiteral"] but instead got "UnaryExpression"
```

### Environment

- @babel/core: 7.14.2

### Possible solution

https://github.com/babel/babel/blob/main/packages/babel-types/src/definitions/typescript.ts#L342
Add `UnaryExpression`.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by colinaaa**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes #13327 
| Patch: Bug Fix?          | 👍
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

Add validator for `TSLiteralType` to accept `UnaryExpression`

eg:

```ts
type A = -1 | 1;
```

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/13525"></a>

### CODE REVIEW

File: packages/babel-types/src/ast-types/generated/index.ts [hunk 1]
```diff
@@ -1905,7 +1905,7 @@ export interface TSMappedType extends BaseNode {
 
 export interface TSLiteralType extends BaseNode {
   type: "TSLiteralType";
-  literal: NumericLiteral | StringLiteral | BooleanLiteral | BigIntLiteral;
+  literal: any;
```
Comment (colinaaa) [inline packages/babel-types/src/ast-types/generated/index.ts:1908]: The automatically generated type does not look right, should I manually correct it?

Comment (JLHwung) [reply packages/babel-types/src/ast-types/generated/index.ts:1908]: This file is generated from `packages/babel-types/scripts/generators/ast-types.js`, we can add a manual allowlist of handwritten types for functional validators.

Comment (colinaaa) [reply packages/babel-types/src/ast-types/generated/index.ts]: I added a `oneOf` allowlist for validator, now it seems good!

**Review by JLHwung — APPROVED** [review]
Thanks!

File: packages/babel-types/src/definitions/typescript.ts [hunk 2]
```diff
@@ -335,12 +336,43 @@ defineType("TSLiteralType", {
...
+            // check operator first
+            unaryOperator(node, "operator", node.operator);
+            unaryExpression(node, "argument", node.argument);
+          } else {
+            // type A = 'foo' | 'bar' | false | 1;
+            literal(parent, key, node);
+          }
+        }
+
+        validator.oneOf = [
```
Comment (colinaaa) [inline packages/babel-types/src/definitions/typescript.ts:365]: should use oneOfNodeType instead

**Review by existentialism — APPROVED** [review]

**Review by nicolo-ribaudo — APPROVED** [review]
Thank you!

**Pull request merged**